### PR TITLE
[FEATURE] Write tests for the Domain Model (#141)

### LINKED ISSUE #141: [FEATURE] Write tests for the Domain Model

### PR DESCRIPTION
**Pull request by tomasnorre**
Resolves #141

### CODE REVIEW

File: Tests/Functional/Domain/Model/ProcessTest.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+/**
+ * Class ProcessTest
+ * @package AOE\Crawler\Tests\Functional\Domain\Model
+ */
+class ProcessTest extends FunctionalTestCase
+{
+    /**
+ * @test
+ */
```
Comment (srotsch) [inline Tests/Functional/Domain/Model/ProcessTest.php]: Indentation looks wrong here

File: Tests/Unit/Domain/Model/ProcessCollectionTest.php [hunk 2]
```diff
@@ -25,27 +25,109 @@
...
+    public function appendThrowsException()
+    {
+        $wrongObjectType = new BackendUtility();
+        $this->subject->append($wrongObjectType);
+    }
+
+    /**
+     * @test
+     */
+    public function append()
```
Comment (srotsch) [inline Tests/Unit/Domain/Model/ProcessCollectionTest.php]: Could you please find a more "speaking" name for this test case?

File: Tests/Unit/Domain/Model/ProcessCollectionTest.php [hunk 3]
```diff
@@ -25,27 +25,109 @@
...
+            $this->subject->offsetGet(100)
+        );
+    }
+
+    /**
+     * @test
+     *
+     * @expectedException \Exception
+     */
+    public function offsetGet()
```
Comment (srotsch) [inline Tests/Unit/Domain/Model/ProcessCollectionTest.php]: Pls. also think about the function name

File: Tests/Unit/Domain/Model/ProcessManagerTest.php [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    /**
+     * Creates the test environment.
+     *
+     * @return void
+     */
+    public function setUp()
+    {
+        $this->subject = new \tx_crawler_domain_process_manager();
+
+        $this->crawlerLibrary = $this->getMock('\tx_crawler_lib', array('dummyMethod'), array(), '', FALSE);
```
Comment (srotsch) [inline Tests/Unit/Domain/Model/ProcessManagerTest.php]: I think we agreed on PSR-2, so lowercase false please

File: Tests/Unit/Domain/Model/ProcessManagerTest.php [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+        define('TYPO3_DOCUMENT_ROOT', '/typo3/document/root/');
+        define('TYPO3_SITE_PATH', '/typo3/site/path/');
+
+        $this->crawlerLibrary->setExtensionSettings(array(
+            'phpPath' => '/path/to/php',
+        ));
+
```
Comment (srotsch) [inline Tests/Unit/Domain/Model/ProcessManagerTest.php]: Remove empty line

File: Tests/Unit/Domain/Model/ProcessManagerTest.php [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    /**
+     * Resets the test environment after the test.
+     *
+     * @return void
+     */
+    public function tearDown()
```
Comment (srotsch) [inline Tests/Unit/Domain/Model/ProcessManagerTest.php]: Not needed, the parent method takes care of unsetting all properties

File: Tests/Unit/Domain/Model/ProcessTest.php [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+            'ttl' => '300',
+            'assigned_items_count' => '20'
+        );
+        $this->subject = new \tx_crawler_domain_process($processObjectArray);
+    }
+
+    /**
+     * @test
+     */
+    public function getActiveReturnInterger()
```
Comment (srotsch) [inline Tests/Unit/Domain/Model/ProcessTest.php]: Return_s_

File: domain/process/class.tx_crawler_domain_process.php [hunk 8]
```diff
@@ -124,10 +128,14 @@ public function countItemsToProcess() {
 	 */
 	public function getProgress() {
 		$all = $this->countItemsAssigned();
-		if ($all > 0) {
-			$res = round((100 / $all) * $this->countItemsProcessed());
-		} else {
-			$res = 0;
+		if ($all<=0){
```
Comment (srotsch) [inline domain/process/class.tx_crawler_domain_process.php]: please add whitespace around comparison operator and between ){

File: domain/process/class.tx_crawler_domain_process.php [hunk 9]
```diff
@@ -124,10 +128,14 @@ public function countItemsToProcess() {
...
+			return 0;
+		}
+
+		$res = round((100 / $all) * $this->countItemsProcessed());
+
+		if($res > 100.0) {
```
Comment (srotsch) [inline domain/process/class.tx_crawler_domain_process.php]: Missing whitespace after if

File: domain/process/class.tx_crawler_domain_process_manager.php [hunk 10]
```diff
@@ -78,6 +78,8 @@ public function __construct() {
 	 * starts multiple processes
 	 *
 	 * @param integer $timeout
+     *
```
Comment (srotsch) [inline domain/process/class.tx_crawler_domain_process_manager.php]: Indentation looks wrong

**Comment by tomasnorre** [comment]
Closed as extend PR was made.

**Pull request closed without merging**